src/canisters/frontend/ic-certified-assets/src/state_machine.rs

```diff
@@ -462,10 +429,12 @@ impl State {
     pub fn delete_asset(&mut self, arg: DeleteAssetArguments) {
         if self.assets.contains_key(&arg.key) {
             for dependent in self.dependent_keys(&arg.key) {
-                let path = AssetPath::from(dependent);
-                self.asset_hashes.delete(path.asset_hash_path_v1().as_vec());
-                self.asset_hashes
-                    .delete(path.asset_hash_path_root_v2().as_vec());
+                self.asset_hashes.remove_responses_for_path(&dependent);
+                self.asset_hashes.remove_responses_for_path_v1(&dependent);
+                if dependent == FALLBACK_FILE {
```

Comment:
Is this a fix that we should mention in the changelog?  Or was this handled elsewhere?

Reply:
Good catch! It may have worked properly with the service worker previously, but it certainly didn't behave properly according to the v2 spec. Added a changelog entry.

I'll still leave this PR as a refactor type since this fix is almost by accident and only a minuscule part of the PR